app/controllers/mileages_controller.rb

```diff
@@ -2,10 +2,13 @@
 
 class MileagesController < ApplicationController
   def index
-    @mileages = current_user.mileages.order(:car_id, meter_status: :asc).page params[:page]
+    @mileages = current_user.mileages.order(:car_id, created_at: :desc).page params[:page]
   end
 
   def show
     @mileage = current_user.mileages.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
```

Comment:
Please remove this. It's a bad way for handling 404 errors. You don't get page status 404 what is useful in tests.
```
Started GET "/mileages/388888" for 127.0.0.1 at 2018-01-09 21:31:28 +0100
Processing by MileagesController#show as HTML
  Parameters: {"id"=>"388888"}
  User Load (0.9ms)  SELECT  "users".* FROM "users" WHERE "users"."deleted_at" IS NULL AND "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 4], ["LIMIT", 1]]
  Mileage Load (0.9ms)  SELECT  "mileages".* FROM "mileages" INNER JOIN "cars" ON "mileages"."car_id" = "cars"."id" WHERE "mileages"."deleted_at" IS NULL AND "cars"."deleted_at" IS NULL AND "cars"."user_id" = $1 AND "mileages"."id" = $2 LIMIT $3  [["user_id", 4], ["id", 388888], ["LIMIT", 1]]
Redirected to http://localhost:3000/mileages
Completed 302 Found in 14ms (ActiveRecord: 1.9ms
```
In your case it's difficult to manage 404 errors. 
To be consequent you should this use in ever 'edit/show/delete' action.
You have ready 404 pages in production which can be customize.